mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp

```diff
@@ -162,6 +162,12 @@ SYCLOpAsmInterface::getAlias(mlir::Type Type, llvm::raw_ostream &OS) const {
            << Ty.getCurrentDimension() << "_";
         return AliasResult::OverridableAlias;
       })
+      .Case<mlir::sycl::MultiPtrType>([&](auto Ty) {
+        OS << "sycl_" << decltype(Ty)::getMnemonic() << "_" << Ty.getDataType()
+           << "_" << mlir::sycl::accessAddressSpaceAsString(Ty.getAddrSpace())
+           << "_" << mlir::sycl::decoratedAccessAsString(Ty.getDecAccess())
+           << "_";
+        return AliasResult::OverridableAlias;
```

Comment:
should this be finalalias?

Reply:
What do others think of reusing the same case as `TupleCopyAssignableValueHolderType`?

Reply:
> should this be finalalias?

Yes, doing it

Reply:
> What do others think of reusing the same case as `TupleCopyAssignableValueHolderType`?

I will revisit this in a future patch maybe.  Are you OK to commit without this change?

Reply:
> What do others think of reusing the same case as `TupleCopyAssignableValueHolderType`?

I feel more inclined to using data type and address space (dropping decorated argument). That way, this would become Overridable again.

Reply:
Why we want to drop the decorated argument?

Reply:
OK, I think I understand why, nm